cachalot/tests/postgres.py

```diff
@@ -91,14 +92,18 @@ def setUp(self):
         self.obj1.save()
         self.obj2.save()
 
+    @all_final_sql_checks
     def test_unaccent(self):
-        Test.objects.create(name='Clémentine')
-        Test.objects.create(name='Clementine')
+        obj1 = Test.objects.create(name='Clémentine')
+        obj2 = Test.objects.create(name='Clementine')
         qs = (Test.objects.filter(name__unaccent='Clémentine')
               .values_list('name', flat=True))
         self.assert_tables(qs, Test)
         self.assert_query_cached(qs, ['Clementine', 'Clémentine'])
+        obj1.delete()
+        obj2.delete()
```

Comment:
I see. Does calling `self.teardown` work?

Reply:
No, I was trying that also as first option, it fails on second pass. Probably because the transaction is above this.
```
======================================================================
FAIL: test_ipv6_address (cachalot.tests.read.ParameterTypeTestCase) [CACHALOT_FINAL_SQL_CHECK = False]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pavel/Projects/Cachalot - benchmark/django-cachalot/cachalot/tests/tests_decorators.py", line 25, in wrapper
    func(self, *args, **kwargs)
  File "/home/pavel/Projects/Cachalot - benchmark/django-cachalot/cachalot/tests/read.py", line 1213, in test_ipv6_address
    self.assert_query_cached(qs, ['2001:db8:0:85a3::ac1f:8001',
  File "/home/pavel/Projects/Cachalot - benchmark/django-cachalot/cachalot/tests/test_utils.py", line 63, in assert_query_cached
    assert_function(data2, result)
AssertionError: Lists differ: ['200[31 chars]:db8:0:85a3::ac1f:8001', '2001:db8:a0b:12f0::1[27 chars]/64'] != ['200[31 chars]:db8:a0b:12f0::1/64']

First differing element 1:
'2001:db8:0:85a3::ac1f:8001'
'2001:db8:a0b:12f0::1/64'

First list contains 2 additional elements.
First extra element 2:
'2001:db8:a0b:12f0::1/64'

+ ['2001:db8:0:85a3::ac1f:8001', '2001:db8:a0b:12f0::1/64']
- ['2001:db8:0:85a3::ac1f:8001',
-  '2001:db8:0:85a3::ac1f:8001',
-  '2001:db8:a0b:12f0::1/64',
-  '2001:db8:a0b:12f0::1/64']
```
I found doing the cleaning by myself is the best option. Maybe you could have some list/set of created objects on the instance and delete them after the test in the decorator, but I find already implemented solution better - but I am open to ideas.